server/models/service-instance.js

```diff
@@ -93,6 +93,15 @@ module.exports = function extendServiceInstance(ServiceInstance) {
   }
   ServiceInstance.recordStatusUpdate = recordStatusUpdate;
 
+  function recordStatusWdUpdate(instanceId, instInfo, callback) {
+    ServiceInstance.findById(instanceId, function(err, instance) {
+      if (err) return callback(err);
+      instance.applicationName = instInfo.appName;
```

Comment:
I'd put an `if(instInfo.appName)` around this, you didn't bump a minor in supervisor, and strong-runner, and pm... which is too much work, so just not nulling applicationName if the supervisor is out of date will be a bit more robust.